### DISCUSSION TITLE
How prevent iMask deleting last character in case of incorrect Date?

### DISCUSSION THREAD

**Discussion by fedorinoGore** [uNmAnNeR/imaskjs#978] [Q&A] [answered]
**Describe the bug**
Im using REact component and when entering non existing date the IMaskInput just deletes the last character in Year value

**To Reproduce**
here is an example
1. Try to enter 29/02/2022 and you will not be able to enter 2022 year, or 2021. Only 2020 will work because this date is valid
2. Try to change 29/02/2020 to 30/02/2020 - you can't do that also.

https://codesandbox.io/s/compassionate-shtern-d5kcmk?file=/src/DateInput.tsx

**Expected behavior**
Just let me turn off this behaviour. 
I want to correct the date or raise an error by myself on my initiative after all data is entered. No correction or validation of the Date required from the iMask package except that created by my own.

**Environment:**
 - OS: OS X
 - Browser: Chrome
 - Version: 119
 - IMask version: react-imask 7.2.0
 - Framework/plugin version if used: React

**Additional context**
It is a pity that library is so badly documented...

**Comment by uNmAnNeR** [comment, marked as answer]
if do not need validations for date just do not use `Date` mask and use common pattern mask instead.

**Reply by fedorinoGore** [reply]
@uNmAnNeR Thanks a lot! That helped.
But now I feel that I'm using the component incorrectly.
I changed the mask to string `"DD{/}MM{/}YYYY"` and this fixed my issue but created new ones.
TS yeels that the type of the mask prop should be `MaskedDate`. 

Another problem is that in `onAccept` event I get masked value no matter if I set `unmask` prop to true/false or 'typed'. And while typing I never get the full value. For example in the input field I type 26/08/1978, but in `onAccept(value: string)` value is **_26/08/197_** until the input looses focus. Only after that the onAccept is called with full value **_26/08/1978_**

Will appreciate any help or code examples...
